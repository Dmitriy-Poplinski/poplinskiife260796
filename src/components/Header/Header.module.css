.header__main {
    height: 6%;
    position: sticky;
    display: flex;
    align-items: center;
    justify-content: center;
};
