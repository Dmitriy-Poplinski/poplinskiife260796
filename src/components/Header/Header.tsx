import React from 'react';

import classes from './Header.module.css';

interface HeaderProps {
    image: string
}

export function Header({ image }: HeaderProps) {
    let color;

    try {
        color = `url(${require(`../../assets/images/${image}.jpg`)})`; 
    } catch (e) {
        color = 'blue'
    };

    return (
        <div 
            className={classes.header__main}
            style={{
                background: color
            }}
            data-testid="header"
        >
            Header
        </div>
    );
}
