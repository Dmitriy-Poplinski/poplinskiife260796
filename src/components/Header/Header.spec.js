import { render, screen } from '@testing-library/react';

import { Header } from './Header';

describe('Header:', () => {

    let data; 

    test('Header gets incorrect string', () => {
        data = 'testing_image';

        render(<Header image={data} />);

        expect(screen.getByTestId('header')).toHaveStyle('background: blue');
        expect(screen.getByTestId('header')).toMatchSnapshot();
    });

    test('Header gets correct string', () => {
        data = 'header_cover';

        render(<Header image={data} />);

        expect(screen.getByTestId('header')).toHaveStyle(`background: url("header_cover.jpg")`);
        expect(screen.getByTestId('header')).toMatchSnapshot();
    });
});
