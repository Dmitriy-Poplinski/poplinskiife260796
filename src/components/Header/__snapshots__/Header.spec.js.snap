// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: Header gets correct string 1`] = `
<div
  class="header__main"
  data-testid="header"
  style="background: url(header_cover.jpg);"
>
  Header
</div>
`;

exports[`Header: Header gets incorrect string 1`] = `
<div
  class="header__main"
  data-testid="header"
  style="background: blue;"
>
  Header
</div>
`;
